x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -751,6 +759,36 @@ function removePreconfiguredConnectorsFromReferences(
   return doc;
 }
 
+// This fixes an issue whereby metrics.alert.inventory.threshold rules had the
+// group for actions incorrectly spelt as metrics.invenotry_threshold.fired vs metrics.inventory_threshold.fired
+function fixInventoryThresholdGroupId(
+  doc: SavedObjectUnsanitizedDoc<RawAlert>
+): SavedObjectUnsanitizedDoc<RawAlert> {
+  const {
+    attributes: { actions },
+  } = doc;
+
+  const updatedActions = actions.map((action) => {
+    // Wrong spelling
+    if (action.group === 'metrics.invenotry_threshold.fired') {
+      return {
+        ...action,
+        group: 'metrics.inventory_threshold.fired',
+      };
+    } else {
+      return action;
+    }
+  });
+
+  return {
+    ...doc,
+    attributes: {
+      ...doc.attributes,
+      actions: [...updatedActions],
```

Comment:
nit: do we need to copy the array here?

```suggestion
      actions: updatedActions,
```